packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -20,6 +20,20 @@ const globalStylesheetsParentElement: Element = document.head || document.body |
 
 let getCustomElement, defineCustomElement, HTMLElementConstructor;
 
+// Read the existing styles from the document
+// This could be enhanced if we associate a hash to every style tag. A unique hash can be generated by the compiler
+// and passed to insertGlobalStylesheet, then set as the data-lwc-ssr value. This way, we would only compare simple attribute
+// values and not the style content.
+(function readExistingStyles() {
+    const styles = globalStylesheetsParentElement.querySelectorAll('style[lwc-scoped-style]');
```

Comment:
do we really need to do this as part of the initialization? or can we wait until the first style needs to be inserted? That will give us more control to only do it when synthetic shadow is used/enabled.

Reply:
Done. This is now happening on first access.

Reply:
I still see loadGlobalStylesheets invoked during evaluation of this module.